clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1111,6 +1112,87 @@ public void testSubscriptionChangesWithAutoCommitDisabled() {
         consumer.close();
     }
 
+    @Test
+    public void testUnsubscribeWillTriggerPartitionsRevokeWithValidGeneration() {
+        Time time = new MockTime();
+        SubscriptionState subscription = new SubscriptionState(new LogContext(), OffsetResetStrategy.EARLIEST);
+        ConsumerMetadata metadata = createMetadata(subscription);
+        MockClient client = new MockClient(time, metadata);
+
+        initMetadata(client, Collections.singletonMap(topic, 1));
+        Node node = metadata.fetch().nodes().get(0);
+
+        ConsumerPartitionAssignor assignor = new RangeAssignor();
```

Comment:
Could we use the new `CooperativeStickyAssignor` for this? It would be good to use an assignor supporting the cooperative protocol, since that's what motivates this PR in the first place

Reply:
Sg